src/Orleans.Runtime/Hosting/DefaultSiloServices.cs

```diff
@@ -104,6 +104,8 @@ internal static void AddDefaultServices(HostBuilderContext context, IServiceColl
             services.AddFromExisting<IProviderManager, IStreamProviderManager>();
             services.TryAddSingleton<GrainRuntime>();
             services.TryAddSingleton<IGrainRuntime, GrainRuntime>();
+            services.TryAddSingleton<GrainCancellationTokenRuntime>();
```

Comment:
why register as a concrete class? Also, these 2 registration will end up as 2 different singletons by the by the way they were registered

Reply:
I just did that same thing as the `GrainRuntime`

Reply:
You uncovered a (non-impacting) bug with GrainRuntime. We shouldn't do that. I checked and there's no usages of resolving the concrete `GrainRuntime` class, so that's why it's not impacting, but we should remove the registration. If we wanted consumers to resolve either of those, we should use the AddFromExisting extension method (but here we should remove the concrete class registration anyway)